code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm

```diff
@@ -0,0 +1,826 @@
+	if (istype(target, /mob/living/simple_animal/hostile/poison/terror_spider/))
+		if (target in enemies)
+			enemies -= target
+		var/mob/living/simple_animal/hostile/poison/terror_spider/T = target
+		if (T.spider_tier > spider_tier)
+			visible_message("<span class='notice'> \icon[src] [src] bows in respect for the terrifying presence of [target] </span>")
+		else if (T.spider_tier == spider_tier)
+			visible_message("<span class='notice'> \icon[src] [src] harmlessly nuzzles [target]. </span>")
+		else if (T.spider_tier < spider_tier && spider_tier >= 4)
+			visible_message("<span class='notice'> \icon[src] [src] gives [target] a stern look. </span>")
+		else
+			visible_message("<span class='notice'> \icon[src] [src] harmlessly nuzzles [target]. </span>")
+		T.CheckFaction()
+		CheckFaction()
+	else if (istype(target, /obj/effect/spider/cocoon))
+		to_chat(src, "Destroying our own cocoons would not help us.")
+	else if (istype(target, /obj/machinery/camera))
+		if (player_breaks_cameras)
+			var/obj/machinery/camera/C = target
+			if (C.status)
+				do_attack_animation(C)
+				C.toggle_cam(src,0)
+				visible_message("<span class='danger'>\the [src] smashes the [C.name].</span>")
+				playsound(src.loc, 'sound/weapons/slash.ogg', 100, 1)
+			else
+				to_chat(src, "The camera is already deactivated.")
+		else
+			to_chat(src, "Your type of spider cannot break cameras.")
+	else if (istype(target, /obj/machinery/door/firedoor))
+		var/obj/machinery/door/firedoor/F = target
+		if (F.density)
+			if (F.blocked)
+				to_chat(src, "The fire door is welded shut.")
+			else
+				visible_message("<span class='danger'>\the [src] pries open the firedoor!</span>")
+				F.open()
+		else
+			to_chat(src, "Closing fire doors does not help.")
+	else if (istype(target, /obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/A = target
+		if (A.density)
+			try_open_airlock(A)
+	else if (ai_type == 2)
+		to_chat(src, "Your current orders forbid you from attacking anyone.")
+	else if (ai_type == 1 && !(target in enemies))
+		to_chat(src, "Your current orders only allow you to defend yourself - not initiate combat.")
+	else if(isliving(target))
+		var/mob/living/L = target
+		if(L.player_logged)
+			to_chat(src, "[target] is braindead, and a waste of our time. (SSD. Server rules prohibit attacking SSDs)")
+			if (L in enemies)
+				enemies -= L
+			return
+		else if (istype(L, /mob/living/silicon/))
+			L.attack_animal(src)
+			return
+		else if(L.reagents && (iscarbon(L)))
+			if (istype(L, /mob/living/carbon/human/))
+				var/mob/living/carbon/human/H = L
+				if(H.dna)
+					if(!(H.species.reagent_tag & PROCESS_ORG) || (H.species.flags & NO_POISON))
+						H.attack_animal(src)
+						return
+			var/inject_target = pick("chest","head") // Yes, there are other body parts. No, the code won't work if we pick any of them. Grrr.
+			if (istype(src, /mob/living/simple_animal/hostile/poison/terror_spider/prince))
```

Comment:
all of these if(istype(src)) conditions need to be 
A) in a seperate proc
B) handled by override